js/foundation.util.box.js

```diff
@@ -74,21 +74,22 @@ function GetDimensions(elem){
   }
 
   var rect = elem.getBoundingClientRect(),
-      parRect = elem.parentNode.getBoundingClientRect(),
+      par = elem.parentNode,
+      parRect = par.getBoundingClientRect(),
       winRect = document.body.getBoundingClientRect(),
       winY = window.pageYOffset,
       winX = window.pageXOffset;
 
   return {
-    width: rect.width,
-    height: rect.height,
+    width: elem.clientWidth,
+    height: elem.clientHeight,
```

Comment:
> The Element.clientWidth property is zero for elements with no CSS or inline layout boxes, otherwise it's the inner width of an element in pixels. It includes padding but not the vertical scrollbar (if present, if rendered), border or margin.
>
>  https://developer.mozilla.org/en-US/docs/Web/API/Element/clientWidth

So border, margin, CSS transformations and zero on inline element. I guess this would introduce some bugs ;)

I would personally use jQuery for these things, as it take care for us of all the low-level and browser specific stuff.

You can also take a look at https://github.com/zurb/foundation-sites/search?q=getBoundingClientRect&type=Issues&utf8=%E2%9C%93 to ensure not to reopen old issues.

Reply:
Not sure but I do not see any error. So .width / height directly? Well, but we would have to migrate this to ES5 then anyway.

Reply:
@JeremyEnglert @kball what solution should we choose here?